lib/smart_name/variants.rb

```diff
@@ -0,0 +1,32 @@
+class SmartName
+  module Variants
+    def simple_key
+      decoded
+        .underscore
+        .gsub(/[^#{OK4KEY_RE}]+/, '_')
+        .split(/_+/)
+        .reject(&:empty?)
+        .map { |key| SmartName.stable_uninflect(key) }
+        .join('_')
+    end
+
+    def url_key
+      @url_key ||= part_names.map do |part_name|
+        stripped = part_name.decoded.gsub(/[^#{OK4KEY_RE}]+/, ' ').strip
+        stripped.gsub(/[\s\_]+/, '_')
+      end * self.class.joint
+    end
+
+    def safe_key
```

Comment:
true, I started with what I think is true about `safe_key`, but I'm not so sure about the specification of the other keys